x/fibre/blob.go

```diff
@@ -0,0 +1,307 @@
+package fibre
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"runtime"
+
+	"github.com/celestiaorg/rsema1d"
+	"github.com/celestiaorg/rsema1d/field"
+)
+
+// ErrBlobTooLarge is returned when the blob size exceeds MaxBlobSize.
+var ErrBlobTooLarge = errors.New("blob size exceeds maximum allowed size")
+
+// Commitment is a commitment to a blob.
+// TODO(@Wondertan): merge with rsema1d.Commitment once it has these methods.
+type Commitment rsema1d.Commitment
+
+// UnmarshalBinary decodes a [Commitment] from bytes.
+func (c *Commitment) UnmarshalBinary(data []byte) error {
+	if len(data) != 32 {
+		return fmt.Errorf("commitment must be 32 bytes, got %d", len(data))
+	}
+	copy(c[:], data)
+	return nil
+}
+
+// String returns the hex-encoded string representation of the commitment.
+func (c Commitment) String() string {
+	return hex.EncodeToString(c[:])
+}
+
+// BlobConfig contains configuration for erasure coding and data handling.
+type BlobConfig struct {
+	// OriginalRows is the number of original rows before erasure coding.
+	// Here we use explicit naming: OriginalRows (K in rsema1d) and ParityRows (N in rsema1d).
+	// SPECDO: The spec uses N to represent total rows (original + parity), while rsema1d defines N as parity only.
+	OriginalRows int
+	// ParityRows is the number of parity rows added by erasure coding.
+	// Total rows = OriginalRows + ParityRows.
+	ParityRows int
+	// RowSizeMin is the minimum row size in bytes.
+	RowSizeMin int
+	// MaxBlobSize is the maximum allowed blob size.
+	MaxBlobSize int
+	// BlobVersion is the version of the row format.
+	BlobVersion uint8
+	// CodingWorkers is the number of workers to use for encoding and decoding rsema1d.
+	CodingWorkers int
+}
+
+// DefaultBlobConfig returns a [BlobConfig] with default values.
+func DefaultBlobConfig() BlobConfig {
+	return BlobConfig{
+		OriginalRows:  1 << 12,   // 4096
+		ParityRows:    3 << 12,   // 12288 (3 * 4096), total rows = 16384
+		RowSizeMin:    1 << 6,    // 64 bytes
+		MaxBlobSize:   128 << 20, // 128 mib
```

Comment:
the comments aren't necessary if we update the constants


```suggestion
		OriginalRows:  4096,
		ParityRows:    12288,
		RowSizeMin:    64,
		MaxBlobSize:   128 * mebibyte,
```

maybe I'm missing something on why these constants use bitwise operations. Is this a type of performance optimization?

Reply:
It just looks cooler 🕶️ 

and I am trying to remember which power of 2 is which number

Reply:
I assume that's sarcasm so let's update to my suggestion and resolve this thread.

Reply:
Can't apply the suggestion as there is no mebibyte literal, but updated manually